scripts/west_commands/sbom/scancode_toolkit_detector.py

```diff
@@ -54,12 +56,14 @@ def detect(data: Data, optional: bool):
     if optional:
         filtered = tuple(filter(lambda file: len(file.licenses) == 0, data.files))
     else:
-        filtered = data.files
+        filtered = tuple(data.files)
```

Comment:
Why you added tuple? In the line 57 is needed to convert interator to actual data. Here we already have list, which is ok for us.

Reply:
I don't need it, I must leave it as some kind of leftover. Removed